samples/asynctests/test_azure_iothub_cli_extension.py

```diff
@@ -335,6 +335,7 @@ def get_target(config):
     return target
 
 def test_iothub_monitor_events(live_iothub_config):
+    pytest.skip('Unable to open connection/socket with the given hostname, skipping for now')
```

Comment:
@annatisch , This test case would raise `RuntimeError: Operation failed` in my current implementation.
So I tested this livetest over uamqp1.2.0 and 1.2.3. I found that this test case actually is always failing, but exceptions are just not raised.

I would prefer skipping it for now.

Here is the log information:
[error information on current implementation.txt](https://github.com/Azure/azure-uamqp-python/files/3737740/error.information.on.current.implementation.txt)
[error information on uamqp1.2.0 and 1.2.3.txt](https://github.com/Azure/azure-uamqp-python/files/3737741/error.information.on.uamqp1.2.0.and.1.2.3.txt)

Reply:
The Az CLI IoT extension uses uamqp for all amqp operations. 

Can we please make sure python uamqp is compatible with the CLI IoT extension use cases and if there are some breaking changes or design change that needs to be updated in the IoT extension you can let me know? In the past @annatisch maintained this partnership (as this test shows). It seems suspicious that with this update the test is skipped.

Interesting datapoint from @yunhaoling that this test is always failing in 1.2.0 and 1.2.3. We are currently using v1.1.0 - we tried updating to 1.2.4  to support Py 3.8 but ran into issues, such as segfaults in Mac environment. We planned to try with 1.2.5 or a later version in the coming days/weeks. 

What changed from 1.1 -> 1.2 that are causing these exceptions?